### ISSUE TITLE
Revisit p2lb protected node revision to ensure proper cleanup

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#6859]
# Problem

With the current protected revision, there is a potential for v2 draft revisions to exist that are "newer" (high id increment). We are currently targeting the protected node revision id and anything _less than_ which leaves these newer revisions around and prevents complete clean up of v2 content.

>Do we want to delete all revisions except for the "current" one? I've run into an issue on brand where things get messy and there are v2 revisions newer than the protected revision. This results in the orphan purger not cleaning things up because the current code only deletes the protected revision and anything that is "less than"

<img width="1154" alt="Screenshot 2023-09-06 at 2 46 11 PM" src="https://github.com/uiowa/uiowa/assets/4663676/232ac891-64be-4969-90e7-2f372e633e27">

_Originally posted by @joewhitsitt in https://github.com/uiowa/uiowa/issues/6830#issuecomment-1708993147_

# Proposed Solution
- [ ] Look for a ways to make sure the protected revision is the newest revision before conversion or change our criteria for cleaning up v2 content after conversion is complete.

# Workaround
- Manually delete those leftover revisions and trigger the `delete orphans` entity_reference_revisions functionality an additional three times to handle paragraph nesting.

**Comment by cory-skeers** [comment]
Couple options could be to 

- On conversion, create a new revision as a copy of the current published one, to set that as protected
- On conversion, get the "true" latest (not latest published) and mark that one as the protected revision
- Leave logic the same as now, but on cleanup, just remove all revisions prior to the latest revision (which should be V3)

**Comment by cory-skeers** [comment]
> On conversion, create a new revision as a copy of the current published one, to set that as protected

After discussing in grooming, I think I'm leaning toward this option, as it gives the possibility of keeping some V3 revisions (up to the 10 limit), which while isn't a necessity, could be nice.

It does bring up the question of which content should be converted: the currently published page or the drafted content? I think we just need to make a choice on which happens here and be consistent, and possibly add some messaging as such.

**Comment by pyrello** [comment]
My hunch is that this is related to the content moderation module, although I'm not totally clear on exactly how yet. I did some debugging to compare what happens when a new revision is created vs when we convert a node and create a new revision based on that. 

https://git.drupalcode.org/project/drupal/-/blob/9.5.10/core/modules/content_moderation/src/EntityOperations.php?ref_type=tags#L173-182

In the above section of code, when saving a new revision, the `if` statement evaluates to true. When converting a node and creating a new revision, the `if` statement evaluates to false.

**Comment by e-marie-w** [comment]
I'm not sure what I'm doing wrong but that if statement is not ever evaluating as true for me. Saving a new v2 revision, a new v3 revision, or creating a new revision as part of the conversion process, it doesn't make a difference. I've now switched to main and it hasn't changed anything. I'm going to continue investigating, but if anyone has any suggestions let me know.

**Comment by pyrello** [comment]
Taking a look to make sure I can duplicate my own work in https://github.com/uiowa/uiowa/issues/6859#issuecomment-1766871697

**Comment by pyrello** [comment]
@e-marie-w brought me up to speed with what has been tried already in #7025 and where it fell short of expectations. I'm going to recreate one of the previous approaches to try to better understand why it didn't work.

**Comment by pyrello** [comment]
There were a couple issues that needed to be addressed:
- The entity was always getting passed in as the default revision (published) regardless of the fact that the view was showing the most recent revision. @e-marie-w started the work of accounting for this by putting together code to pull the most recent revision and swap it out for the entity that gets passed in. All I did was move this process in the Converter class.
- When actually processing the entity, we were loading the paragraphs using the `target_id` instead of their `target_revision_id` from the referencing entity. I think that Paragraphs must work similarly to how the base entity works, where if you pull the version by the id, you get the current published version, not the most recent revision. To get the most recent revision, we needed to switch to using the `target_revision_id`.

After making these changes, the conversion appropriately targets the most recent revision of every entity that we are working with through the process.

I used HMB as my test site. When running the cleanup command, I was running into issues where the process would not complete because it complains there are still entities that prevent the V2/P2LB config from being uninstalled. For the portion that is supposed to be removing the orphaned paragraphs, I see this output:
```
>  [notice] Checked Paragraph revisions for orphans: 49 of 49 in 0 sec (0 deleted)
>  [notice] Message: Paragraph: Deleted 0 revisions (0 entities) in 0 sec.
> 
>  [notice] Checked Paragraph revisions for orphans: 49 of 49 in 0 sec (0 deleted)
>  [notice] Message: Paragraph: Deleted 0 revisions (0 entities) in 0 sec.
> 
>  [notice] Checked Paragraph revisions for orphans: 49 of 49 in 0 sec (0 deleted)
>  [notice] Message: Paragraph: Deleted 0 revisions (0 entities) in 0 sec.
```

I tried it with faculty-senate.uiowa.edu and it worked without any issues. Next step is to try to understand what's different between the two sites.
